# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: ef817155231f
# Full Hash: ef817155231f6fed522ce793cbc82b7be019de56
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Constify conduit getters. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D122244
# ==============================================================================

diff -r 310fdf39b051 -r ef817155231f dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Sun Apr 18 21:25:11 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Tue Aug 10 14:58:34 2021 +0200
@@ -137,7 +137,7 @@
    */
   bool SendRtcp(const uint8_t* data, size_t len) override;
 
-  bool HasCodecPluginID(uint64_t aPluginID) override { return false; }
+  bool HasCodecPluginID(uint64_t aPluginID) const override { return false; }
 
   void DeliverPacket(rtc::CopyOnWriteBuffer packet, PacketType type) override;
 
@@ -155,10 +155,10 @@
    */
   bool SetLocalSSRCs(const std::vector<uint32_t>& aSSRCs,
                      const std::vector<uint32_t>& aRtxSSRCs) override;
-  std::vector<uint32_t> GetLocalSSRCs() override;
+  std::vector<uint32_t> GetLocalSSRCs() const override;
   bool SetRemoteSSRC(uint32_t ssrc, uint32_t rtxSsrc) override;
   bool UnsetRemoteSSRC(uint32_t ssrc) override { return true; }
-  bool GetRemoteSSRC(uint32_t* ssrc) override;
+  bool GetRemoteSSRC(uint32_t* ssrc) const override;
   bool SetLocalCNAME(const char* cname) override;
   bool SetLocalMID(const std::string& mid) override;
 
@@ -240,7 +240,7 @@
   // Accessed only on the Call thread.
   bool mDtmfEnabled;
 
-  Mutex mMutex;
+  mutable Mutex mMutex;
 
   // Call worker thread. All access to mCall->Call() happens here.
   const RefPtr<AbstractThread> mCallThread;